content/en/apps/guides/integrations/best-practices/rapidpro.md

```diff
@@ -0,0 +1,108 @@
+- Regarding SMS costs from experience, zero-rated messages are recommended so that respondents do not incur charges.
+
+### Flow design
+Tips and best practices are listed below
+ - Length of the survey.
+   - If possible, keep the questions to between 5 and 10. Long surveys may either lead to low completion rates or rushed responses that affects the data quality.
+- Types of questions
+   - Close ended questions are better and easier to respond and handle in RapidPro since respondents only have to send a number or letter such as 1 for “Yes”, 2 for “No”. Avoid sensitive questions since privacy cannot be guaranteed where shared phones are common.
+- Intro and outro messages
+   - These serve the purpose of giving the survey details such as background of the survey, number of questions, data protection, whether there shall be follow up, SMS billing, thank you notes etc. Include these if possible.
+- Consent and opt out
+   - Include questions that give the respondent an opportunity to opt in or out of the survey. If they opt out, do not send a follow up text.
+- SMS length
+   - Keep it short, to the size of one SMS (160 characters). Longer messages may not display well on the recipients’ devices.
+
+### Flow programming
+- Utilize skip logic to ask relevant questions only e.g. number of children if only they have children.
+- Automate error messages to to validate responses e.g. You submitted an invalid response. Reply 1 for “Yes”, 2 for “No”. These can be customized.
+- Use reminders and follow-up messages selectively on respondents by first differentiating those that completed a flow versus those that did not. This reduces spamming.
+- Customize messages as much as possible by pre-populating information already known such as name, to target respondents, especially when phones are shared among clients.
+- Translate all messages, especially when deploying in a multi-lingual environment. This ensures that respondents fully understand the survey in their language.
+- Beware of the timing of the surveys that directly affects response rates. From experience, sending questions when respondents are busy with their errands during the day ultimately leads to low response rates as opposed to evenings when they are done for the day.
+- Make sure you handle unsolicited responses by redirecting such to for example, a flow that eventually alerts concerned individuals e.g. reports of an outbreak etc.
+
+### Configuration
+1. Persist a unique identifier for each respondent entering the flows. This identifier shall be used to link flow runs to the recipients to make data analysis smooth.
+2. Use consistent language and design patterns.
```

Comment:
what do you mean here by consistent design pattenrs? consistent messaging patterns?

Reply:
Message design. For example, if a contact name comes at the beginning of a message, keep it that way all through.